code/game/world.dm

```diff
@@ -160,6 +160,33 @@ GLOBAL_PROTECT(security_mode)
 	else
 		to_chat(world, "<span class='boldannounce'>Rebooting world...</span>")
 		Master.Shutdown()	//run SS shutdowns
+
+	if(SERVER_TOOLS_PRESENT)
+		var/do_hard_reboot
+		// check the hard reboot counter
+		var/ruhr = CONFIG_GET(number/rounds_until_hard_restart)
+		switch(ruhr)
+			if(-1)
+				do_hard_reboot = FALSE
+			if(0)
+				do_hard_reboot = TRUE
+			else
+				var/counter_path = "data/round_counter.txt"

```

Comment:
It should save to this file on shutdown, and read and delete it on world start, that way crashes and what not reset the count.